TTS/tts/utils/speakers.py

```diff
@@ -154,15 +155,21 @@ def save_d_vectors_to_file(self, file_path: str) -> None:
         """
         self._save_json(file_path, self.d_vectors)
 
-    def set_d_vectors_from_file(self, file_path: str) -> None:
+    def set_d_vectors_from_file(self, file_path: str, data: List = None) -> None:
         """Load d_vectors from a json file.
 
         Args:
             file_path (str): Path to the target json file.
         """
         self.d_vectors = self._load_json(file_path)
-        speakers = sorted({x["name"] for x in self.d_vectors.values()})
-        self.speaker_ids = {name: i for i, name in enumerate(speakers)}
+
+        # load speakers from data, because during the training we can just use some speakers from d_vector_file
+        if data is not None:
+            self.speaker_ids, _ = self.parse_speakers_from_data(data)
+        else:
+            speakers = sorted({x["name"] for x in self.d_vectors.values()})
+            self.speaker_ids = {name: i for i, name in enumerate(speakers)}
```

Comment:
I don't understand the use-case

Reply:
the `data` is the items from the dataset. We need to print to the user only the speakers used in training and not all those present in d_vectors. With this speaker_ids filter, you will only have the speakers that will actually be used in training, otherwise, it will be confusing for the user because he will see all the speakers present in d_vectors and not actually only those who have samples in the training. And so it would be impossible for the user to know if the data loader's speaker filtering is in fact working.  Do you still think a new function is needed? whenever we call set_d_vectors_from_file we would have to call this function... Why not keep in this function?

Reply:
Done